modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -254,142 +256,10 @@ class ReplCalculator(
         )
     }
 
-    def execute(sender: ActorRef, er: ExecuteRequest): Unit = {
-      val (outputType, newCode) = er.code match {
-        case resolverRegex(r) =>
-          log.debug("Adding resolver: " + r)
-          val (logR, resolver) = CustomResolvers.fromString(r)
-          resolvers = resolver :: resolvers
-          (`text/plain`, s""" "Resolver added: $logR!" """)
-
-        case repoRegex(r) =>
-          log.debug("Updating local repo: " + r)
-          repo = new File(r.trim)
-          repo.mkdirs
-          (`text/plain`, s""" "Repo changed to ${repo.getAbsolutePath}!" """)
-
-        case dpRegex(local, cp) =>
-          log.debug(s"Fetching ${if(local == "l") "locally" else ""} deps using repos: " + resolvers.mkString(" -- "))
-          val tryDeps = Deps.script(cp, resolvers, repo)
-
-          tryDeps match {
-            case TSuccess(deps) =>
-              eval( """sparkContext.stop() """)("CP reload processed successfully",
-                (str: String) => "Error in :dp: \n%s".format(str)
-              )
-              val (_r, replay) = repl.addCp(deps)
-              _repl = Some(_r)
-              preStartLogic()
-              replay()
-              val newJarList = if (local == "l") {
-                  "Nil"
-                } else {
-                  deps.map(x => x.replaceAll("\\\\", "\\\\\\\\")).mkString("List(\"", "\",\"", "\")")
-                }
-              (`text/html`,
-                s"""
-                   |//updating deps
-                   |globalScope.jars = ($newJarList ::: globalScope.jars.toList).distinct.toArray
-                   |//restarting spark
-                   |reset()
-                   |globalScope.jars.toList
-                 """.stripMargin
-              )
-            case TFailure(ex) =>
-              log.error(ex, "Cannot add dependencies")
-              (`text/html`, s""" <p style="color:red">${ex.getMessage}</p> """)
-          }
-
-        case cpRegex(cp) =>
-          val jars = cp.trim().split("\n").toList.map(_.trim()).filter(_.length > 0)
-          repl.evaluate( """sparkContext.stop()""")._1 match {
-            case Failure(str) =>
-              log.error("Error in :cp: \n%s".format(str))
-            case _ =>
-              log.info("CP reload processed successfully")
-          }
-          val (_r, replay) = repl.addCp(jars)
-          _repl = Some(_r)
-          preStartLogic()
-          replay()
-          val newJarList = jars.map(x => x.replaceAll("\\\\", "\\\\\\\\")).mkString("List(\"", "\",\"", "\")")
-          (`text/html`,
-            s"""
-              |//updating deps
-              |globalScope.jars = ($newJarList ::: globalScope.jars.toList).distinct.toArray
-              |//restarting spark
-              |reset()
-              |globalScope.jars.toList
-            """.stripMargin
-          )
+    private var commandInterpreters = combineIntepreters(command_interpreters.defaultInterpreters)
```

Comment:
this might later include dynamically loaded command interpreters too, e.g. `conf.getConfig("notebook.kernel.interpreters.custom")`